native/src/main/java/io/ballerina/lib/data/csvdata/utils/CsvUtils.java

```diff
@@ -0,0 +1,373 @@
+package io.ballerina.lib.data.csvdata.utils;
+
+import io.ballerina.runtime.api.TypeTags;
+import io.ballerina.runtime.api.flags.SymbolFlags;
+import io.ballerina.runtime.api.types.ArrayType;
+import io.ballerina.runtime.api.types.Field;
+import io.ballerina.runtime.api.types.IntersectionType;
+import io.ballerina.runtime.api.types.RecordType;
+import io.ballerina.runtime.api.types.TupleType;
+import io.ballerina.runtime.api.types.Type;
+import io.ballerina.runtime.api.utils.StringUtils;
+import io.ballerina.runtime.api.utils.TypeUtils;
+import io.ballerina.runtime.api.utils.ValueUtils;
+import io.ballerina.runtime.api.values.BArray;
+import io.ballerina.runtime.api.values.BDecimal;
+import io.ballerina.runtime.api.values.BMap;
+import io.ballerina.runtime.api.values.BString;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.ballerina.lib.data.csvdata.utils.Constants.SKIP_LINE_RANGE_SEP;
+
+/*
+ * Utility functions used in the CSV data module.
+ * 
+ * @since 0.1.0
+ */
+public class CsvUtils {
+    private static final long[] EMPTY_LONG_ARRAY =  new long[]{};
+
+    public static void validateExpectedArraySize(int size, int currentSize) {
+        if (size != -1 && size > currentSize) {
+            throw DiagnosticLog.error(DiagnosticErrorCode.INVALID_EXPECTED_ARRAY_SIZE, currentSize);
+        }
+    }
+
+    public static boolean isExpectedTypeIsArray(Type expectedType) {
+        expectedType = TypeUtils.getReferredType(expectedType);
+
+        return switch (expectedType.getTag()) {
+            case TypeTags.TUPLE_TAG, TypeTags.ARRAY_TAG -> true;
+            default -> false;
+        };
+    }
+
+    public static boolean isBasicType(Type type) {
+        return switch (type.getTag()) {
+            case TypeTags.INT_TAG, TypeTags.STRING_TAG, TypeTags.BOOLEAN_TAG, TypeTags.DECIMAL_TAG, TypeTags.FLOAT_TAG,
+                 TypeTags.NULL_TAG, TypeTags.JSON_TAG, TypeTags.ANYDATA_TAG, TypeTags.UNION_TAG,
+                 TypeTags.INTERSECTION_TAG, TypeTags.CHAR_STRING_TAG, TypeTags.BYTE_TAG, TypeTags.SIGNED8_INT_TAG,
+                 TypeTags.SIGNED16_INT_TAG, TypeTags.SIGNED32_INT_TAG, TypeTags.UNSIGNED8_INT_TAG,
+                 TypeTags.UNSIGNED16_INT_TAG, TypeTags.UNSIGNED32_INT_TAG -> true;
+            default -> false;
+        };
+    }
+
+    public static String[] createHeadersForParseLists(BArray csvElement, String[] headers, CsvConfig config) {
```

Comment:
Is there a reason why this method takes `header` array modify it and then return it. I think it can either create it's own array and return it to the caller or not return anything since caller anyway has the array. I personally prefer the first option